packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -221,7 +221,11 @@ class AnsiTerminal {
   String clearScreen() => supportsColor ? _clear : '\n\n';
 
   set singleCharMode(bool value) {
-    stdin.lineMode = !value;
+    try {
+      stdin.lineMode = !value;
+    } on IOException {
+      // This can throw for some terminals; we ignore the error.
```

Comment:
The specific error we're seeing here is "Inappropriate ioctl for device" right?

Why would we see this? Normally that error is a bad sign (indicating that we're reading the errno code inappropriately).